lib/server_sent_event_stage.ex

```diff
@@ -52,79 +53,123 @@ defmodule ServerSentEventStage do
   @doc false
   def handle_info(:connect, state) do
     url = compute_url(state)
-    {:ok, id} = connect_to_url(url, state.headers)
-    {:noreply, [], %{state | id: id}}
+    state = do_connect(url, state)
+    {:noreply, [], state}
   end
 
-  def handle_info(%HTTPoison.AsyncStatus{id: id, code: code}, %{id: id} = state) do
-    state = %{state | state: :connected}
+  def handle_info(message, %{conn: conn} = state) when conn != nil do
+    case HTTP.stream(state.conn, message) do
+      {:ok, conn, responses} ->
+        state = %{state | conn: conn}
+        {state, events} = Enum.reduce_while(responses, {state, []}, &handle_mint_response/2)
 
-    state =
-      if code == 200 do
-        Logger.debug(fn -> "#{__MODULE__} connected" end)
-        state
-      else
-        Logger.warn(fn -> "#{__MODULE__} unexpected status: #{code}" end)
-        do_refresh(state)
-      end
+        {:noreply, events, state}
+
+      {:error, conn, error, responses} ->
+        state = %{state | conn: conn}
+        ref = state.ref
+        {state, events} = Enum.reduce_while(responses, {state, []}, &handle_mint_response/2)
+        {_, {state, events}} = handle_mint_response({:error, ref, error}, {state, events})
+
+        {:noreply, events, state}
+
+      :unknown ->
+        handle_unknown_info(message, state)
+    end
+  end
+
+  def handle_info(message, state) do
+    handle_unknown_info(message, state)
+  end
 
+  def handle_unknown_info({data_tag, _, _}, state) when data_tag in [:ssl, :tcp] do
+    # The can occur after we've re-connected: drop them on the floor
     {:noreply, [], state}
   end
 
-  def handle_info(%HTTPoison.AsyncHeaders{id: id}, %{id: id} = state) do
+  def handle_unknown_info({closed_tag, _}, state) when closed_tag in [:ssl_closed, :tcp_closed] do
```

Comment:
Can these still happen? I was under the impression this was something to do with `:hackney`, and that was one of the benefits of using `Mint`.

Reply:
What happens is that the the `ref` associated with the connection changes, and so we get messages from the old port before it's fully closed. It's not ideal, but the advantage over `hackney` is that we're getting the messages (so we can ignore them), instead of the messages getting into the Hackeny internals where we can't do anything about it.